patches/master_patch.patch

```diff
@@ -2004,10 +2004,38 @@ index df954bc8f81850101676b1c20f688ba0ba6b4b09..4683b2853bb467ec9f58616ede207d45
  
      // Non-standard APIs
 diff --git a/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp b/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
-index b80b134d9db6ebee55af88010de2718ed46d1860..cd1a7d38170bc705f882ef824d24f070ca7c2255 100644
+index b80b134d9db6ebee55af88010de2718ed46d1860..eb50c91fb2d7235f70398345eabe712c349870ca 100644
 --- a/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
 +++ b/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
-@@ -3905,6 +3905,10 @@ void HTMLMediaElement::unlockUserGesture() {
+@@ -459,6 +459,7 @@ HTMLMediaElement::HTMLMediaElement(const QualifiedName& tagName,
+       m_playingRemotely(false),
+       m_inOverlayFullscreenVideo(false),
+       m_mostlyFillingViewport(false),
++      m_hasPlayed(false),
+       m_audioTracks(this, AudioTrackList::create(*this)),
+       m_videoTracks(this, VideoTrackList::create(*this)),
+       m_textTracks(this, nullptr),
+@@ -2261,7 +2262,7 @@ Nullable<ExceptionCode> HTMLMediaElement::play() {
+     if (isGestureNeededForPlayback()) {
+       // If we're already playing, then this play would do nothing anyway.
+       // Call playInternal to handle scheduling the promise resolution.
+-      if (!m_paused) {
++      if (!m_paused || m_hasPlayed) {
```

Comment:
no, it is actually blocked on muon side.
When users start playing video, it will not block. But as user drag timeline , some sites like vimeo or netflix will trigger a `pause()` first, and that will cause `m_paused` to be true and it won't continue playing.